client/state/plugins/wporg/reducer.js

```diff
@@ -48,7 +79,37 @@ export function items( state = {}, action ) {
 	}
 }
 
+export function lists( state = defaultPluginsListState, action ) {
+	const { type, category, page, data } = action;
+	switch ( type ) {
+		case WPORG_PLUGIN_RECEIVE_LIST:
+			return Object.assign( {}, state, updatePluginsList( defaultPluginsListState, category, page, data ) );
```

Comment:
I often see this pattern, and it's one of the first things that might come to mind when we are creating Redux reducers, but it's actually perpendicular to what we want to be using Redux for. This mangles the different pieces of data together when we want nice separation of data.

Imagine that instead of having one updater which updates five pieces of data that we had five updaters each updating their own simple piece of data in response to this same type.

```js
const isFetching = ( state = {}, { type, category } ) => get( {
	[ WPORG_PLUGIN_RECEIVE_LIST ]: { ...state, [ category ]: false },
	[ WPORG_PLUGIN_FETCH_LIST ]: { ...state, [ category ]: true },
}, type, state );

const lastFetchedPage = ( state = {}, { type, category, page } ) => get( {
	[ WPORG_PLUGINS_RECEIVE_LIST ]: { ...state, [ category ]: page },
}, type, state );

export default combineReducers(
	isFetching,
	lastFetchedPage,
);
```

I only included a couple of the simple reducers, but hopefully you can see how we're removing the logic about those different data pieces from big reducers and splitting them into several small reducers.